openest/generate/functions.py

```diff
@@ -927,16 +934,20 @@ class AuxiliaryResultApplication(calculation.Application):
     """
     Perform both main and auxiliary calculation, and order as main[0], aux, main[1:]
     """
-    def __init__(self, region, subapp_main, subapp_aux):
+    def __init__(self, region, subapp_main, subapp_aux, keeplastonly):
         super(AuxiliaryResultApplication, self).__init__(region)
         self.subapp_main = subapp_main
         self.subapp_aux = subapp_aux
+        self.keeplastonly = keeplastonly
 
     def push(self, ds):
         for yearresult in self.subapp_main.push(ds):
             for yearresult_aux in self.subapp_aux.push(ds):
                 next # Just take the last one
```

Comment:
Not strictly in your PR but this `next` statement wont do anything. Needs to be `next()`?

Reply:
I believe it is written correctly. The goal here is to do nothing within this inner for loop, just allowing it to run.

Reply:
I guess I've never seen `next` as a statement and I can't find it documented in core python docs but it seems to work so 👍 . Nevermind!

Marking as resolved.

Reply:
Actually, the python statement I wanted is called `pass`. `next` was working for the wrong reasons. I've swapped it over.